server/public/model/config.go

```diff
@@ -4322,49 +4322,110 @@ func (o *Config) GetSanitizeOptions() map[string]bool {
 	return options
 }
 
-func (o *Config) Sanitize() {
-	if o.LdapSettings.BindPassword != nil && *o.LdapSettings.BindPassword != "" {
-		*o.LdapSettings.BindPassword = FakeSetting
+type SanitizeConfigRules struct {
+	Contains   []string
+	StartsWith []string
+	EndsWith   []string
+}
+
+func sanitizeConfigMap(config map[string]interface{}, rules SanitizeConfigRules) *AppError {
```

Comment:
the idea is recursively iterate over the config map validating all the keys, checking if it is either a string or an array of strings and if the key name matches the sanitization rules.